packages/rollup-contracts/test/rollup-list/StateCommitmentChain.spec.ts

```diff
@@ -0,0 +1,377 @@
+
+    it('should throw if submitting an empty batch', async () => {
+      const emptyBatch = []
+      await TestUtils.assertRevertsAsync(
+        'Cannot submit an empty state commitment batch',
+        async () => {
+          await stateChain.appendStateBatch(emptyBatch)
+        }
+      )
+    })
+
+    it('should add to batches array', async () => {
+      await stateChain.appendStateBatch(DEFAULT_STATE_BATCH)
+      const batchesLength = await stateChain.getBatchesLength()
+      batchesLength.toNumber().should.equal(1)
+    })
+
+    it('should update cumulativeNumElements correctly', async () => {
+      await stateChain.appendStateBatch(DEFAULT_STATE_BATCH)
+      const cumulativeNumElements = await stateChain.cumulativeNumElements.call()
+      cumulativeNumElements.toNumber().should.equal(DEFAULT_STATE_BATCH.length)
+    })
+
+    it('should calculate batchHeaderHash correctly', async () => {
+      const localBatch = await appendAndGenerateBatch(DEFAULT_STATE_BATCH)
+      const expectedBatchHeaderHash = await localBatch.hashBatchHeader()
+      const calculatedBatchHeaderHash = await stateChain.batches(0)
+      calculatedBatchHeaderHash.should.equal(expectedBatchHeaderHash)
+    })
+
+    it('should add multiple batches correctly', async () => {
+      const numBatches = 5
+      for (let batchIndex = 0; batchIndex < numBatches; batchIndex++) {
+        const cumulativePrevElements = DEFAULT_STATE_BATCH.length * batchIndex
+        const localBatch = await appendAndGenerateBatch(
+          DEFAULT_STATE_BATCH,
+          batchIndex,
+          cumulativePrevElements
+        )
+        const expectedBatchHeaderHash = await localBatch.hashBatchHeader()
+        const calculatedBatchHeaderHash = await stateChain.batches(batchIndex)
+        calculatedBatchHeaderHash.should.equal(expectedBatchHeaderHash)
+      }
+      const cumulativeNumElements = await stateChain.cumulativeNumElements.call()
+      cumulativeNumElements
+        .toNumber()
+        .should.equal(numBatches * DEFAULT_STATE_BATCH.length)
+      const batchesLength = await stateChain.getBatchesLength()
+      batchesLength.toNumber().should.equal(numBatches)
+    })
+
+    it('should throw if submitting more state commitments than number of txs in canonical tx chain', async () => {
+      const numBatches = 5
+      for (let i = 0; i < numBatches; i++) {
+        await stateChain.appendStateBatch(DEFAULT_STATE_BATCH)
+      }
+      await TestUtils.assertRevertsAsync(
+        'Cannot append more state commitments than total number of transactions in CanonicalTransactionChain',
+        async () => {
+          await stateChain.appendStateBatch(DEFAULT_STATE_BATCH)
+        }
+      )
+    })
+  })
+
+  describe('verifyElement() ', async () => {
+    it('should return true for valid elements for different batches and elements', async () => {
+      await appendTxBatch(DEFAULT_TX_BATCH)
+      await appendTxBatch(DEFAULT_TX_BATCH)
+      const numBatches = 3
```

Comment:
A small improvement of coverage here would be to make the batches distinct, e.g. `batches = [abatch, adifferentbatch, adifferentdifferentbatch]` and using `for (let batch in batches)`.  Just takes care of the edge case where we somehow verify against the previous batch, but since they are the same the inclusion proof still passes.